chain-core/src/init/coin.rs

```diff
@@ -5,23 +5,22 @@
 
 use crate::init::{MAX_COIN, MAX_COIN_DECIMALS, MAX_COIN_UNITS};
 use crate::state::tendermint::TendermintVotePower;
+use crate::state::tendermint::TENDERMINT_MAX_VOTE_POWER;
 use parity_codec::{Decode, Encode, Input};
 
-use crate::state::tendermint::TENDERMINT_MAX_VOTE_POWER;
-#[cfg(feature = "serde")]
-use serde::de::{Deserialize, Deserializer, Error, Visitor};
-#[cfg(feature = "serde")]
-use serde::Serialize;
+use serde::{Deserialize, Serialize, Serializer};
```

Comment:
please keep the feature guards for serde:
```
#[cfg(feature = "serde")]
```

Reply:
ok
i also add unit test

Reply:
done